src/RelayPricer.ts

```diff
@@ -0,0 +1,28 @@
+import { SourceApi } from './types/RelayPricer';
+
+export default class RelayPricer {
+    private readonly sourceApi: SourceApi;
+
+    constructor(sourceApi: SourceApi) {
+        this.sourceApi = sourceApi;
+    }
+
+    async getPrice(
+        sourceCurrency: string,
+        targetCurrency: string,
+        intermediateCurrency?: string
+    ): Promise<number> {
+        const intermediary = intermediateCurrency
+            ? intermediateCurrency
+            : 'USD';
+
+        const [sourcePrice, targetPrice] = await Promise.all([
+            this.sourceApi.query(sourceCurrency, intermediary),
+            this.sourceApi.query(targetCurrency, intermediary)
+        ]);
+        if (sourcePrice === 0 || targetPrice === 0) {
+            throw new Error('price cannot be zero');
```

Comment:
the message is a bit meaningless for the caller of the method I'd say

Reply:
updated

Reply:
Conversation continues in https://github.com/rsksmart/rif-relay-client/pull/28#discussion_r956041187